src/modules/user/user.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable, UnprocessableEntityException } from '@nestjs/common';
+import { Prisma } from '@prisma/client';
+import { PrismaService } from '../prisma/prisma.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+
+@Injectable()
+export class UserService {
+  constructor(private prismaService: PrismaService) {}
+  async findAll() {
+    return await this.prismaService.user.findMany();
+  }
+
+  async create(createUser: CreateUserDto) {
+    const existUser = await this.prismaService.user.findFirst({
+      where: { email: createUser.email },
+    });
+
+    if (existUser) {
+      throw new UnprocessableEntityException('Email is already exist.');
+    }
+
+    const userPayload: Prisma.UserCreateInput = {
+      email: createUser.email,
+      name: createUser.name,
```

Comment:
`...createUser` will not throw exception when `nickname` is present. Because Prisma doesn't trip field that isn't exist in Database
```
[Nest] 89398  - 09/03/2021, 3:28:32 PM   ERROR [ExceptionsHandler] 
Invalid `prisma.user.create()` invocation:

{
  data: {
    email: 'linh@gmail.com',
    name: 'Linh Pham',
    nickname: 'Nick name',
    ~~~~~~~~
    profile: {
      create: {
        nickname: 'Nick name'
      }
    }
  }
}

Unknown arg `nickname` in data.nickname for type UserCreateInput. Did you mean `name`? Available args:
type UserCreateInput {
  email: String
  name: String
  createdAt?: DateTime
  updatedAt?: DateTime
  posts?: PostCreateNestedManyWithoutAuthorInput
  profile: ProfileCreateNestedOneWithoutUserInput
}
```